monai/transforms/post/array.py

```diff
@@ -122,94 +122,108 @@ class AsDiscrete(Transform):
     Args:
         argmax: whether to execute argmax function on input data before transform.
             Defaults to ``False``.
-        to_onehot: whether to convert input data into the one-hot format.
-            Defaults to ``False``.
-        num_classes: the number of classes to convert to One-Hot format.
+        to_onehot: if not None, convert input data into the one-hot format with specified number of classes.
```

Comment:
thanks, could you add a few lines of examples for different use cases?

Reply:
Sure, added examples in the latest commit.
Could you please help review it again?

Thanks in advance.